ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/ChainHeadTracker.java

```diff
@@ -69,33 +86,79 @@ public CompletableFuture<BlockHeader> getBestHeaderFromPeer(final EthPeer peer)
         .setMessage("Requesting chain head info from {}...")
         .addArgument(peer::getLoggableId)
         .log();
-    final CompletableFuture<AbstractPeerTask.PeerTaskResult<List<BlockHeader>>>
-        bestHeaderFromPeerCompletableFuture = getBestHeaderFromPeerCompletableFuture(peer);
-    final CompletableFuture<BlockHeader> future = new CompletableFuture<>();
-    bestHeaderFromPeerCompletableFuture.whenComplete(
-        (peerResult, error) -> {
-          if (peerResult != null && !peerResult.getResult().isEmpty()) {
-            final BlockHeader chainHeadHeader = peerResult.getResult().get(0);
-            peer.chainState().update(chainHeadHeader);
-            future.complete(chainHeadHeader);
-            LOG.atDebug()
-                .setMessage("Retrieved chain head info {} from {}...")
-                .addArgument(
-                    () -> chainHeadHeader.getNumber() + " (" + chainHeadHeader.getBlockHash() + ")")
-                .addArgument(peer::getLoggableId)
-                .log();
-          } else {
-            LOG.atDebug()
-                .setMessage("Failed to retrieve chain head info. Disconnecting {}... {}")
-                .addArgument(peer::getLoggableId)
-                .addArgument(error != null ? error : "Empty Response")
-                .log();
-            peer.disconnect(
-                DisconnectMessage.DisconnectReason.USELESS_PEER_FAILED_TO_RETRIEVE_CHAIN_HEAD);
-            future.complete(null);
-          }
-        });
-    return future;
+
+    if (synchronizerConfiguration.isPeerTaskSystemEnabled()) {
+      return ethContext
+          .getScheduler()
+          .scheduleServiceTask(
+              () -> {
+                GetHeadersFromPeerTask task =
+                    new GetHeadersFromPeerTask(
+                        Hash.wrap(peer.chainState().getBestBlock().getHash()),
+                        0,
+                        1,
+                        0,
+                        Direction.FORWARD,
+                        protocolSchedule);
+                PeerTaskExecutorResult<List<BlockHeader>> taskResult =
+                    peerTaskExecutor.executeAgainstPeer(task, peer);
+                if (taskResult.responseCode() == PeerTaskExecutorResponseCode.SUCCESS
+                    && taskResult.result().isPresent()) {
```

Comment:
No, it's not possible to have success with an empty result. This is just here to avoid IDE warnings about accessing the content of the result without checking if it's present first. I can remove it and disable those checks instead?